packages/@lwc/engine-core/src/framework/register-stylesheets.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isUndefined } from '@lwc/shared';
+import { Template } from './template';
+import { TemplateStylesheetFactories } from './stylesheet';
+import { flattenStylesheets } from './utils';
+import { checkVersionMismatch } from './check-version-mismatch';
+
+function checkStylesheetsVersionMismatch(stylesheets: TemplateStylesheetFactories | undefined) {
+    if (!isUndefined(stylesheets)) {
+        for (const stylesheet of flattenStylesheets(stylesheets)) {
+            checkVersionMismatch(stylesheet, 'stylesheet');
+        }
+    }
+}
```

Comment:
I moved version mismatch checking here, from `registerTemplate`. As it turns out, stylesheet mismatch checking was not working anyway, because `registerTemplate` is called before the stylesheets are attached to the `tmpl` object.